libraries/standard/mqtt/src/mqtt.c

```diff
@@ -26,6 +26,12 @@
 #include "mqtt_state.h"
 #include "private/mqtt_internal.h"
 
+
+/* Default value for the CONNACK receive retries. */
+#ifndef MQTT_CONNACK_RECEIVE_RETRY_COUNT
+    #define MQTT_CONNACK_RECEIVE_RETRY_COUNT    ( 5U )
```

Comment:
Should this raise an **error** instead of using the default value? The main reason being that default values can go unnoticed by the user..

Reply:
This config is not always used. It is only required if the API is called with 0 timeout value. We switch to retries only if we find it out at run time. I don't think raising an error would be a good idea for handling a run time behavior, hence used default value